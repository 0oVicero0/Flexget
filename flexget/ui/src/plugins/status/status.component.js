/* global angular */
(function () {
    'use strict';

    angular
        .module('plugins.status')
        .component('statusView', {
            templateUrl: 'plugins/status/status.tmpl.html',
            controllerAs: 'vm',
            controller: statusController
        });

    function statusController($scope, $filter, statusService) {
        var vm = this;
        
        vm.$onInit = activate;
        vm.timeSorter = timeSorter;

        vm.tableData = {
            data: [],
            'table-row-id-key': 'id',
            'column-keys': [
                'name',
                'last_execution.start',
                'last_execution.produced',
                'last_execution.accepted',
                'last_execution.rejected',
                'last_execution.failed'
            ]
        }

        // Needs to use $scope, mdDataTable takes private scopes, so using vm doesn't work        
        $scope.getSuccess = getSuccessValue;
        
        function getSuccessValue(rowId) {
            var value = $filter('filter')(vm.tableData.data, { id: rowId })[0];
            return value.last_execution.succeeded;
        }

        function timeSorter(a, b) {
            return new Date(b) - new Date(a);
        }

        function activate() {
            getStatus();
        }

        function getStatus() {
            statusService.getStatus()
                .then(setStatuses)
                .cached(setStatuses);
        }
        
        function setStatuses(response) {
            vm.tableData.data = response.data;
        }
    }
}());