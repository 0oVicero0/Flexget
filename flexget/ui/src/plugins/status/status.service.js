/* global angular */
(function () {
    'use strict';

    angular
        .module('plugins.status')
        .factory('statusService', statusService);

    function statusService($http, exception) {
        return {
            getStatus: getStatus
        };

        function getStatus() {
            return $http.get('/api/status/', { etagCache: true })
                .catch(callFailed);
        }

        function callFailed(error) {
            return exception.catcher(error);
        }
    }
}());