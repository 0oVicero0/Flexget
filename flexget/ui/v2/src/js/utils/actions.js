export const createAction = namespace => (type, payload, {
  ignore = false,
  message,
} = {}) => ({
  type,
  payload,
  error: (payload instanceof Error),
  meta: {
    ignore,
    message,
    namespace,
  },
});

export const createLoading = namespace => type => ({
  type,
  meta: {
    namespace,
    loading: true,
  },
});
