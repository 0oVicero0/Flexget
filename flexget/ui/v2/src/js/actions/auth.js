import * as utils from 'utils/actions';
import { get, post } from 'utils/fetch';

export const AUTH = 'AUTH';
export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';

const createAction = utils.createAction(AUTH);
const loading = utils.createLoading(AUTH);

export function checkLogin() {
  return (dispatch) => {
    get('/server/version')
      .then(() => dispatch(createAction(LOGIN)))
      .catch(() => {});
  };
}

export function login(credentials) {
  return (dispatch) => {
    dispatch(loading(LOGIN));
    post('/auth/login', credentials)
      .then(() => dispatch(createAction(LOGIN)))
      .catch(err => dispatch(createAction(LOGIN, err)));
  };
}

export function logout() {
  return (dispatch) => {
    dispatch(loading(LOGOUT));
    get('/auth/logout')
      .then(() => dispatch(createAction(LOGOUT)))
      .then(err => dispatch(createAction(LOGOUT, err)));
  };
}
