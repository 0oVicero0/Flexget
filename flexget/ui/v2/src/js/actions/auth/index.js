import { createAction, loading } from 'utils/actions';
import { get, post } from 'utils/fetch';

const PREFIX = '@flexget/auth/';
export const LOGIN = `${PREFIX}LOGIN`;
export const LOGOUT = `${PREFIX}LOGOUT`;

export function checkLogin() {
  return dispatch => get('/server/version')
      .then(() => dispatch(createAction(LOGIN)))
      .catch(() => {});
}

export function login(credentials) {
  return (dispatch) => {
    dispatch(loading(LOGIN));
    return post('/auth/login', credentials)
      .then(() => dispatch(createAction(LOGIN)))
      .catch(err => dispatch(createAction(LOGIN, err)));
  };
}

export function logout() {
  return (dispatch) => {
    dispatch(loading(LOGOUT));
    return post('/auth/logout')
      .then(() => dispatch(createAction(LOGOUT)))
      .catch(err => dispatch(createAction(LOGOUT, err)));
  };
}
