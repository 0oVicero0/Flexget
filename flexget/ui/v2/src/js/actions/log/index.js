import oboe from 'oboe';
import { createAction } from 'utils/actions';

const PREFIX = '@flexget/log/';
export const LOG_START = `${PREFIX}LOG_START`;
export const LOG_MESSAGE = `${PREFIX}LOG_MESSAGE`;
export const LOG_DISCONNECT = `${PREFIX}LOG_DISCONNECT`;
export const LOG_LINES = `${PREFIX}LOG_LINES`;
export const LOG_QUERY = `${PREFIX}LOG_QUERY`;
export const LOG_CLEAR = `${PREFIX}LOG_CLEAR`;

export function startLogStream() {
  return (dispatch, getState) => {
    const { lines, query } = getState().log;
    const stream = oboe({
      url: `api/server/log?lines=${lines}&search=${query}`,
      method: 'GET',
    });

    const promise = new Promise((resolve, reject) => {
      stream
        .on('start', () => dispatch(createAction(LOG_START)))
        .on('node', '{message}', message => dispatch(createAction(LOG_MESSAGE, message)))
        .done(() => resolve())
        .fail(err => reject(err));
    });

    promise.abort = () => {
      stream.abort();
      dispatch(createAction(LOG_DISCONNECT));
    };

    return promise;
  };
}

export function setLines(lines) {
  return createAction(LOG_LINES, lines);
}

export function setQuery(query) {
  return createAction(LOG_QUERY, query);
}

export function clearLogs() {
  return createAction(LOG_CLEAR);
}

