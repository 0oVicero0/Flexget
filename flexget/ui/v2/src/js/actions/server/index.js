import { createAction, loading } from 'utils/actions';
import { post } from 'utils/fetch';

const PREFIX = '@flexget/server/';
export const SERVER_RELOAD = `${PREFIX}SERVER_RELOAD`;
export const SERVER_SHUTDOWN = `${PREFIX}SERVER_SHUTDOWN`;


function manageServer(action, operation) {
  return (dispatch) => {
    dispatch(loading(action));
    return post('/server/manage', { operation })
      .then(({ data }) => dispatch(createAction(action, {}, { message: data.message })))
      .catch(err => dispatch(createAction(action, err)));
  };
}

export function reloadServer() {
  return manageServer(SERVER_RELOAD, 'reload');
}

export function shutdownServer() {
  return manageServer(SERVER_SHUTDOWN, 'shutdown');
}
