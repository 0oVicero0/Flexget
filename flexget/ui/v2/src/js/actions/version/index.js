import * as utils from 'utils/actions';
import { get } from 'utils/fetch';

export const VERSION = 'VERSION';
export const GET_VERSION = 'GET_VERSION';

const createAction = utils.createAction(VERSION);
const loading = utils.createLoading(VERSION);

export function getVersion() {
  return (dispatch) => {
    dispatch(loading(GET_VERSION));
    return get('/server/version')
      .then(version => dispatch(createAction(GET_VERSION, version)))
      .catch(err => dispatch(createAction(GET_VERSION, err)));
  };
}
