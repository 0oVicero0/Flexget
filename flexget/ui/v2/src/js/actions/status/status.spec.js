import { closeStatus } from 'actions/status';

describe('actions/status', () => {
  it('should create an action to close the status', () => {
    expect(closeStatus()).toMatchSnapshot();
  });
});
