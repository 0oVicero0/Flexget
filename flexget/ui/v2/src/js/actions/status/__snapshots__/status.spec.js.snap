// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions/status should create an action to close the status 1`] = `
Object {
  "type": "CLOSE_STATUS",
}
`;
